core-methods/src/main/java/com/mercadopago/sdk/android/coremethods/domain/usecase/GenerateCardTokenUseCase.kt

```diff
@@ -8,23 +8,50 @@ import com.mercadopago.sdk.android.coremethods.domain.utils.Result
 import com.mercadopago.sdk.android.coremethods.ui.components.textfield.INT_TWO
 
 internal const val EXPIRATION_YEAR_START = "20"
+internal const val EXPIRATION_YEAR_MIN_LENGTH = 2
+internal const val SECURITY_CODE_MIN_LENGTH = 3
 
+@Suppress("ReturnCount")
 internal class GenerateCardTokenUseCase(
     private val repository: CoreMethodsRepository,
 ) {
     suspend operator fun invoke(
         cardNumber: String,
-        expirationDate: String,
         securityCode: String,
+        expirationDate: String?,
     ): Result<CardToken, ResultError> {
-        val expirationMonth = expirationDate.ifEmpty { "0" }.take(INT_TWO).toInt()
-        val expirationYear =
-            (EXPIRATION_YEAR_START + expirationDate.ifEmpty { "0" }.takeLast(INT_TWO)).toInt()
+        val expirationDateIsNotNull = expirationDate != null
+
+        if (cardNumber.isEmpty()) {
+            return Result.Error(ResultError.Validation("card id number cannot be empty"))
+        }
+
+        if (securityCode.isEmpty()) {
+            return Result.Error(ResultError.Validation("security code cannot be empty"))
+        }
+
+        if (securityCode.length < SECURITY_CODE_MIN_LENGTH) {
+            return Result.Error(ResultError.Validation("security code length cannot be smaller than tree"))
+        }
```

Comment:
there are some cards for which the security code might be optional. we can maybe try to do something like this:

```suggestion
        if (securityCode.isNotEmpty() && securityCode.length < SECURITY_CODE_MIN_LENGTH) {
            return Result.Error(ResultError.Validation("security code length cannot be smaller than tree"))
        }
```